django/contrib/admindocs/views.py

```diff
@@ -411,8 +412,9 @@ def simplify_regex(pattern):
     """
     pattern = replace_named_groups(pattern)
     pattern = replace_unnamed_groups(pattern)
-    # clean up any outstanding regex-y characters.
-    pattern = pattern.replace('^', '').replace('$', '').replace('?', '')
+    # Remove any outstanding regex metacharacters.
+    pattern = re.sub(r'[?^$+*]', r'', pattern)
```

Comment:
What if any of the new special characters appear escaped, like if the url has a literal `+`?

Reply:
Good thought. This was probably already broken in the past when it just did `.replace('?', '')`, etc. But we can certainly improve it...